JoonasHeinonen/escape-from-zordoom-prison
# Feature/check point

### [Checkpoint system](https://trello.com/c/moSwtxNS)

https://user-images.githubusercontent.com/33086486/232335803-c440aa8f-23a0-4f4c-838c-d29fa4c94e58.mp4

**Changes:**
- Once the player moves activates the checkpoint. 
- Later one the player dies.
- The level restarts and the player is back to the checkpoint.
- When player is dead, a fade in animation is played, node `FadeIn`.
- The `PlayerUI` should be hidden once the player is dead.

**How to test:**
 - pull down the checkpoint branch
 - Go to the door and get the notification that you are at the checkpoint.
 - After that go to the edge of the level and fall off.
 - The level Should fade to black and reload to the last point the player was at.
 - See that the `PlayerUI` node is hidden on player death as well.

**Sources cited:**
 - https://www.youtube.com/watch?v=YraYIe-q0uo
 - [Seperate child from parent without affecting position set](https://godotengine.org/qa/133522/seperate-child-from-parent-without-affecting-position-set)
